src/v2/Components/Cells/PartnerCell.tsx

```diff
@@ -41,6 +41,9 @@ const PartnerCell: React.FC<PartnerCellProps> = ({
   const locations = extractNodes(partner.locationsConnection)
   const meta = uniq(locations.map(location => location.city?.trim())).join(", ")
   const image = partner.profile?.image?.cropped
+  const badgedPartners = partner.categories
+    ?.filter(category => category?.slug === "black-owned")
+    .map(category => ({ children: category?.name }))
```

Comment:
**suggestion [non-blocking]:** use more intermediate variables

I think what this is doing is taking the categories array, finding whether it contains the black owned PartnerCategory and transforming it into an object that is suitable for the `EntityHeader` component, right?

I might suggest some intermediate vars - something more like this:

```ts
// coerce away the case of nil by falling back to an empty array
const categories = partner.categories || []
// supporting a new category badge would mean updating this list of slugs:
const badgedCategorySlugs = ["black-owned"]
const matchingCategories = categories.filter(category => badgedCategorySlugs.includes(category.slug))
const badges = matchingCategories.map(category => ({ children: category.name }))
```

And I guess typing this out has me wondering if there should be a function extracted here with tests.

*format via [Conventional Comments](https://conventionalcomments.org/)*

Reply:
If you were going to go the route of testing this as a more formal thing, I'd definitely just do this in Metaphysics; since this is just some temporary logic if it's in Force.

Reply:
Should we cut out a separate ticket for a potential MP breakout? Or should the above int variable breakdown go in on its own?

I personally think both should be done.

Reply:
Added in the suggested variables and also created the following JIRA ticket for a MP breakout of the filtering logic:
https://artsyproduct.atlassian.net/browse/GRO-829